### ISSUE TITLE
[Clients] Stop passing --binary to sed in the yugabyte-clients post-install script.

### ISSUE THREAD

**Issue by kevincox** [yugabyte/yugabyte-db#11914]
### Description

`--binary` has no effect on POSIX systems and doesn't exist on some implementations of sed (notably busybox). This causes the `sed` substitutions to fail. "Luckily" the rest of the script ignores the failure and appears to mostly work. We should just remove the flag for better compatibility.

**Comment by m-iancu** [comment]
@kevincox is this done after https://github.com/yugabyte/yugabyte-cloud/pull/4135 ? If not can you elaborate a bit on which yugabyte-db scripts use this flag?

**Comment by kevincox** [comment]
No, https://github.com/yugabyte/yugabyte-cloud/pull/4135 is a very ugly hack to remove this "in post". We need to remove this from the source script and then we can remove that hacky workaround.
